File: examples/basics.py
import imgproc
from imgproc import *


# open a webcam to take pictures
camera = Camera(320, 240)

# Open a viewer window to display images
viewer = Viewer(320, 240, "The first step")

# take a picture from the camera
img = camera.grabImage()

# display the image in the viewer
viewer.displayImage(img)


# iterate over each pixel in the image
for x in range(0, img.width):
	for y in range(0, img.height):
		# Get the value at the xth column and yth row, place the intensities into variables
		red, green, blue = img[x, y]

		# test to find which colour is dominant
		if red > green and red > blue:
			# Red is dominant, so set the pixel to completely red
			img[x, y] = 255, 0, 0
		elif green > red and green > blue:
			img[x, y] = 0, 255, 0
		else:
			# if red or green isn't dominant, blue must be
			img[x, y] = 0, 0, 255


# display the image again
viewer.displayImage(img)

# delay for 5000 milliseconds to give us time to see the changes, then exit
waitTime(5000)


# end of the script

File: examples/basics.c
#include "imgproc.h"


int main(int argc, char * argv[])
{
	// initialise the library
	init_imgproc();


	// open the webcam, with a capture resolution of width 320 and height 240
	Camera * cam = camOpen(160, 120);

	// create a new viewer of the same resolution with a caption
	Viewer * view = viewOpen(160, 120, "First");

	// capture an image from the webcam	
	Image * img = camGrabImage(cam);

	// display the image in the viewer
	viewDisplayImage(view, img);


	// iterators
	unsigned int x, y;
	for(x = 0; x < img->width; x++){
		for(y = 0; y < img->height; y++){
			// Get a pointer to the current pixel
			char * pixel = imgGetPixel(img, x, y);

			// extract the red, green and blue intensities
			// index 0 is blue and 2 is red
			char red = pixel[2];
			char green = pixel[1];
			char blue = pixel[0];

			// test for predominantly red
			if(red > green && red > blue){
				// set the pixel to the colour 255, 0, 0 (completely red)
				imgSetPixel(img, x, y, 255, 0, 0);
			}
			// green
			else if(green > red && green > blue){
				imgSetPixel(img, x, y, 0, 255, 0);
			}
			// assume blue otherwise
			else {
				imgSetPixel(img, x, y, 0, 0, 255);
			}
		}
	}

	// display the image to view the changes
	viewDisplayImage(view, img);

	// wait for 5 seconds to give us a chance to see the changes
	waitTime(5000);

	// now we will free the memory for the various objects
	imgDestroy(img);
	viewClose(view);
	camClose(cam);

	// finally we unintialise the library
	quit_imgproc();
	return 0;
}
